### PR TITLE
xtensa: define arch_proc_id() for xtensa

### PR DESCRIPTION
**Pull request by nashif**
This interface is currently defined in the soc. Make it more generic and define it on the architecture level so we can use it elsewhere.

- xtensa: add arch_proc_id()
- intel_adsp: use arch_proc_id() instead of local function

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

**Comment by marc-hb** [comment]
Funny how many tests pass while some others don't compile

We all know about "DLL Hell", what is corresponding name for .h files?

- #41543

```
c /__w/zephyr/zephyr/arch/arc/core/offsets/offsets.c
In file included from ../../../../../../include/zephyr/arch/cpu.h:12,
                 from ../../../../../../include/zephyr/kernel_includes.h:33,
                 from ../../../../../../include/zephyr/kernel.h:17,
                 from /__w/zephyr/zephyr/arch/arc/core/offsets/offsets.c:25:
../../../../../../include/zephyr/sys/arch_interface.h:436:24: error: 'arch_proc_id' declared 'static' but never defined [-Werror=unused-function]
  436 | static inline uint32_t arch_proc_id(void);
      |                        ^~~~~~~~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

**Comment by nashif** [comment]
> Funny how many tests pass while some others don't compile

the issue is clear, this is only a multi-core thing so it works fine on non-smp, other SMP architectures are looking for the implementation of the same interface, which I did not put in, this might end up as an xtensa thing after all.

**Review by andyross — DISMISSED** [review]
Makes sense.  Regarding naming: we've never been consistent as to whether arch_*() means "the specified Zephyr architecture interface" or "anything owned by the arch layer".  Various xtensa arch functions have been using an xtensa_ prefix for things like this.   Probably not worth trying to audit for consistency, but just something to be aware of.

**Review by ceolin — DISMISSED** [review]

File: include/zephyr/arch/xtensa/arch_inlines.h [hunk 1]
```diff
@@ -32,6 +32,14 @@ static ALWAYS_INLINE _cpu_t *arch_curr_cpu(void)
 	return cpu;
 }
 
+static ALWAYS_INLINE uint32_t arch_proc_id(void)
```
[the remark this answers is not in the document]
Comment (nashif) [reply include/zephyr/arch/xtensa/arch_inlines.h:35]: Right, thats why I added the extra comment in there and emphasized it is a placeholder for architectures that do not support or have  it yet.

**Review by andyross — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**